src/Mobs/Monster.cpp

```diff
@@ -300,7 +300,12 @@ void cMonster::Tick(std::chrono::milliseconds a_Dt, cChunk & a_Chunk)
 		{
 			if (--m_GiveUpCounter == 0)
 			{
-				FinishPathFinding();
+				FinishPathFinding();  // Not to be confused with StopMovingToPosition, this just discards the current path and calculates another.
```

Comment:
Got a name which means "Either finish or restart, depending on whether we're done or not"?